src/g_statusbar/shared_sbar.cpp

```diff
@@ -1010,7 +1012,8 @@ void DBaseStatusBar::DrawCrosshair ()
 	}
 	int health = Scale(CPlayer->health, 100, CPlayer->mo->GetDefault()->health);
 
-	ST_DrawCrosshair(health, viewwidth / 2 + viewwindowx, viewheight / 2 + viewwindowy, CrosshairSize);
+	const double size = PrevCrosshairSize * (1.0 - ticFrac) + CrosshairSize * ticFrac;
```

Comment:
Friendly reminder that `interpolatedvalue()` exists :)

https://github.com/ZDoom/gzdoom/blob/8311e77a26e104606660576605a633bba2e960b2/src/common/utility/vectors.h#L1611-L1615